### DISCUSSION TITLE
Не работает ConversationalRetrievalChain

### DISCUSSION THREAD

**Discussion by Falconwatch** [ai-forever/gigachain#31] [Q&A]
Использую пример: https://github.com/ai-forever/gigachain/blob/master/docs/extras/integrations/retrievers/wikipedia.ipynb 

на строке result = qa({"question": question, "chat_history": chat_history}) падает ошибка 

ResponseError: (URL('https://ngw.devices.sberbank.ru:9443/api/v2/oauth'), 400, b'{"code":4,"message":"Can\'t decode \'Authorization\' header"}', Headers([('server', 'nginx'), ('date', 'Mon, 09 Oct 2023 14:53:20 GMT'), ('content-type', 'application/json'), ('content-length', '58'), ('connection', 'keep-alive'), ('vary', 'Origin'), ('vary', 'Access-Control-Request-Method'), ('vary', 'Access-Control-Request-Headers'), ('cache-control', 'no-cache, no-store, max-age=0, must-revalidate'), ('pragma', 'no-cache'), ('expires', '0'), ('x-content-type-options', 'nosniff'), ('strict-transport-security', 'max-age=31536000 ; includeSubDomains'), ('x-frame-options', 'DENY'), ('x-xss-protection', '1 ; mode=block'), ('referrer-policy', 'no-referrer'), ('allow', 'POST'), ('strict-transport-security', 'max-age=31536000; includeSubDomains')]))

Что делать?

**Comment by mklyazhev** [comment]
Не знаю насколько еще актуально, но у меня была такая проблема, и дело было в том, что в credentials для GigaChat надо подавать именно "авторизационные данные", а не Client Secret или токен доступа.
Авторизационные данные можно получить во время генерации клиентского ключа (это поле так и называется) или перевести сам Client Secret в base64. 
Надеюсь кому-то поможет.
